src/service/resources-impl.js

```diff
@@ -850,39 +853,41 @@ export class Resources {
         }
       }
 
-      if (minTop != -1) {
+      if (minTop < Infinity) {
         this.setRelayoutTop_(minTop);
       }
 
       // Execute scroll-adjusting resize requests, if any.
       if (scrollAdjSet.length > 0) {
         this.vsync_.run({
           measure: state => {
-            state./*OK*/scrollHeight = this.viewport_./*OK*/getScrollHeight();
             state./*OK*/scrollTop = this.viewport_./*OK*/getScrollTop();
           },
           mutate: state => {
-            let minTop = -1;
+            let minTop = Infinity;
+            let diff = 0;
             scrollAdjSet.forEach(request => {
               const box = request.resource.getLayoutBox();
-              minTop = minTop == -1 ? box.top : Math.min(minTop, box.top);
+              minTop = Math.min(minTop, box.top);
+              diff += request.newHeight - box.height;
```

Comment:
Actually, this should only diff if the `box.bottom` is above the current `scrollTop`.